appservice/src/deploy/localGitDeploy.ts

```diff
@@ -45,17 +65,29 @@ export async function localGitDeploy(client: SiteClient, fsPath: string): Promis
             }
             return undefined;
             // tslint:disable-next-line:no-unsafe-any
-        } else if (err.message.indexOf('error: failed to push') >= 0) {
-            const forcePush: vscode.MessageItem = { title: localize('forcePush', 'Force Push') };
-            const pushReject: string = localize('localGitPush', 'Push rejected due to Git history diverging.');
-            await ext.ui.showWarningMessage(pushReject, forcePush, DialogResponses.cancel);
-            // tslint:disable-next-line:no-floating-promises
-            localGit.push(remote, 'HEAD:master', { '-f': true });
         } else {
             throw err;
         }
     }
 
-    ext.outputChannel.appendLine(formatDeployLog(client, (localize('localGitDeploy', `Deploying Local Git repository to "${client.fullName}"...`))));
-    await waitForDeploymentToComplete(client, kuduClient, commitId);
+    async function pushAndWaitForDeploymentToComplete(token?: vscode.CancellationToken, forcePush: boolean = false): Promise<void> {
+        await new Promise((resolve: () => void, reject: (error: Error) => void): void => {
```

Comment:
I would create a new cancellation token source inside `pushAndWaitForDeploymentToComplete` every time. I don't think it's optional because either way `localGit.push` could throw an error